### PR TITLE
Adding modules over quotient and local rings to Docu

### PR DESCRIPTION
**Pull request by wdecker**

### CODE REVIEW

File: docs/src/CommutativeAlgebra/ModulesOverMultivariateRings/free_modules.md [hunk 1]
```diff
@@ -15,30 +15,30 @@ Pages = ["free_modules.md"]
...
     Canonical maps such us the canonical projection onto a quotient module arise in many 
     constructions in commutative algebra. The `FreeMod` type is designed so that it allows
     for the caching of such maps when executing functions. The `direct_sum` function discussed
     in this section provides an example.
 
-## Constructor
+## Constructors
 
 ```@docs
 free_module(R::MPolyRing, n::Int, name::String = "e"; cached::Bool = false)
```
Comment (HechtiDerLachs) [inline docs/src/CommutativeAlgebra/ModulesOverMultivariateRings/free_modules.md:44]: Below in line 53 there is a discussion of `dim(F)` as a synonym to `rank(F)` for a free module `F`. But the dimension of an arbitrary module can also be understood as the dimension of its support, right? Shouldn't we reserve the `dim` command for that?

**Review by HechtiDerLachs — APPROVED** [review]
Looks good to me.

One remark: I broke with the `SRow` paradigm for the `coordinates` function of modules over localized rings. They return a `Vector` rather than an `SRow`. At the time, I found this more natural and it was also suggested to me by other Oscar developers. 

This does not seem to play a role here, but I wanted to give the heads up. This was also mentioned in the list of things to still be addressed for the modules over `MPolyQuo`s; see the corresponding issue.

**Comment by wdecker** [comment]
@HechtiDerLachs 

julia> coordinates(x*FR[1])
Sparse row with positions [1] and values fmpq_mpoly[x]


julia> coordinates(RL(x)*FRL[1])
Sparse row with positions [1] and values MPolyLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyComplementOfPrimeIdeal{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}[x//1]


julia> coordinates(RQ(x)*FRQ[1])
Sparse row with positions [1] and values MPolyQuoElem{fmpq_mpoly}[x]


julia> coordinates(RQL(x)*FRQL[1])
Sparse row with positions [1] and values MPolyQuoLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyComplementOfPrimeIdeal{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}[x//1]

File: docs/src/CommutativeAlgebra/ModulesOverMultivariateRings/free_modules.md (hunk 1 above)
Comment (wdecker) [reply docs/src/CommutativeAlgebra/ModulesOverMultivariateRings/free_modules.md:44]: Frankly, I find `dim` here a little strange as well, but it is available in UngradedModules.jl. @jankoboehm ???

**Comment by HechtiDerLachs** [comment]
> @HechtiDerLachs
> 
> julia> coordinates(x*FR[1]) Sparse row with positions [1] and values fmpq_mpoly[x]
> 
> julia> coordinates(RL(x)*FRL[1]) Sparse row with positions [1] and values MPolyLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyComplementOfPrimeIdeal{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}[x//1]
> 
> julia> coordinates(RQ(x)*FRQ[1]) Sparse row with positions [1] and values MPolyQuoElem{fmpq_mpoly}[x]
> 
> julia> coordinates(RQL(x)*FRQL[1]) Sparse row with positions [1] and values MPolyQuoLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyComplementOfPrimeIdeal{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}[x//1]

Thanks for checking. Indeed, it seems, I did everything right for the modules. But for the ideals we have different outcomes, see below.
```julia
julia> R, (x, y) = PolynomialRing(QQ, ["x", "y"]);

julia> U = complement_of_ideal(ideal(R, [x]))
complement of ideal(x)

julia> L, _ = localization(R, U);

julia> I = ideal(L, [x])
ideal in localization of Multivariate Polynomial Ring in x, y over Rational Field at the complement of ideal(x) generated by the elements x//1

julia> coordinates(x^2, I)
[x//1]

julia> typeof(ans)
AbstractAlgebra.Generic.MatSpaceElem{MPolyLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyComplementOfPrimeIdeal{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}}

```
Continuing with the above setup we also find
```julia
julia> FL = FreeMod(L, 1)
Free module of rank 1 over localization of Multivariate Polynomial Ring in x, y over Rational Field at the complement of ideal(x)

julia> coordinates(FL[1])
Sparse row with positions [1] and values MPolyLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyComplementOfPrimeIdeal{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}[1//1]

julia> M, _ = sub(FL, [y*FL[1]])
(Submodule with 1 generator
1 -> y//1*e[1]
represented as subquotient with no relations., Map with following data
Domain:
=======
M
Codomain:
=========
FL)

julia> coordinates(FL[1], M)
Sparse row with positions [1] and values MPolyLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyComplementOfPrimeIdeal{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}[1//y]

julia> coordinates(x*M[1])
Sparse row with positions [1] and values MPolyLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyComplementOfPrimeIdeal{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}[x//1]

julia> FR = FreeMod(R, 1)
Free module of rank 1 over Multivariate Polynomial Ring in x, y over Rational Field

julia> coordinates(FR[1])
Sparse row with positions [1] and values fmpq_mpoly[1]

julia> M, _ = sub(FR, [x*FR[1]])
(Submodule with 1 generator
1 -> x*e[1]
represented as subquotient with no relations., Map with following data
Domain:
=======
M
Codomain:
=========
FR)

julia> coordinates(x^2*FR[1], M)
Sparse row with positions [1] and values fmpq_mpoly[x]
```
Now with a different multiplicative set:
```julia

julia> S = powers_of_element(x)
powers of fmpq_mpoly[x]

julia> W, _ = localization(R, S)
(localization of Multivariate Polynomial Ring in x, y over Rational Field at the powers of fmpq_mpoly[x], Map with following data
Domain:
=======
Multivariate Polynomial Ring in x, y over Rational Field
Codomain:
=========
localization of Multivariate Polynomial Ring in x, y over Rational Field at the powers of fmpq_mpoly[x])

julia> FW = FreeMod(W, 1);

julia> coordinates(y*FW[1])
Sparse row with positions [1] and values MPolyLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyPowersOfElement{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}[y//1]

julia> M, _ = sub(FW, [y*FW[1]])
(Submodule with 1 generator
1 -> y//1*e[1]
represented as subquotient with no relations., Map with following data
Domain:
=======
M
Codomain:
=========
FW)

julia> coordinates(y^2*FW[1], M)
Sparse row with positions [1] and values MPolyLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyPowersOfElement{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}[y//1]
```
Localization of quotient rings also seems to be consistent. Continuing with the above: 
```julia
julia> I = ideal(R, [x^2 + y^2])
ideal(x^2 + y^2)

julia> Q, _ = quo(R, I)
(Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal(x^2 + y^2), Map from
Multivariate Polynomial Ring in x, y over Rational Field to Q defined by a julia-function with inverse)

julia> QL, _ = localization(Q, U)
(Localization of Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal(x^2 + y^2) at the multiplicative set complement of ideal(x), Map from
Q to Localization of Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal(x^2 + y^2) at the multiplicative set complement of ideal(x) defined by a julia-function)

julia> FQL = FreeMod(QL, 1)
Free module of rank 1 over Localization of Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal(x^2 + y^2) at the multiplicative set complement of ideal(x)

julia> M, _ = sub(FQL, [x*FQL[1]])
(Submodule with 1 generator
1 -> 0
represented as subquotient with no relations., Map with following data
Domain:
=======
M
Codomain:
=========
FQL)

julia> coordinates(x^2*FQL[1], M)
Sparse row with positions Int64[] and values MPolyQuoLocalizedRingElem{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly, MPolyComplementOfPrimeIdeal{FlintRationalField, fmpq, FmpqMPolyRing, fmpq_mpoly}}[]


julia> iszero(M)
true
```
So, it seems to be mainly a decision on the ideals that we need to take at some point. Sorry for spamming this PR.

File: docs/src/CommutativeAlgebra/ModulesOverMultivariateRings/free_modules.md (hunk 1 above)
Comment (jankoboehm) [reply docs/src/CommutativeAlgebra/ModulesOverMultivariateRings/free_modules.md:44]: It is only defined for free modules

**Pull request merged**